packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -406,9 +400,46 @@ - (void)updatePlayingState {
   _displayLink.running = _isPlaying || self.waitingForFrame;
 }
 
+- (void)sendFailedToLoadVideoEvent:(NSError *)error {
+  if (!NSThread.isMainThread) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [self sendFailedToLoadVideoEvent:error];
+    });
+    return;
+  }
+  if (_eventSink == nil) {
+    return;
+  }
+  __block NSString *message = @"Failed to load video";
+  void (^add)(NSString *) = ^(NSString *detail) {
+    if (detail != nil) {
+      message = [message stringByAppendingFormat:@": %@", detail];
+    }
+  };
+  NSError *underlyingError = error.userInfo[NSUnderlyingErrorKey];
+  // add more details to error message
+  // https://github.com/flutter/flutter/issues/56665
+  add(error.localizedDescription);
+  add(error.localizedFailureReason);
+  add(underlyingError.localizedDescription);
+  add(underlyingError.localizedFailureReason);
+  _eventSink([FlutterError errorWithCode:@"VideoError" message:message details:nil]);
+}
+
 - (void)setupEventSinkIfReadyToPlay {
   if (_eventSink && !_isInitialized) {
     AVPlayerItem *currentItem = self.player.currentItem;
+    // if status is Failed here then this was called from onListenWithArguments which means
+    // _eventSink was nil when was called observeValueForKeyPath with updated Failed status
+    // and error was not sent and never will be so it is needed to send it here to have
+    // future returned by initialize resolved instead of in state of infinite waiting,
```

Comment:
Can you separate these 4 lines into multiple sentences? I have a hard time understanding this.